### PR TITLE
Endtoend cluster improvements

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Some niceties and convenience functionality in endtoend/cluster setup:

- `NewBareCluster`: create a cluster without assuming any of its components are actually used. e.g. you may create a cluster with no tablets, and skip checking for tablet version.
- `MysqlctlProcess` to support `Connect`, returning a valid connection
- `MysqlctlProcess` to return some paths of interest.

## Related Issue(s)

~~This PR extends https://github.com/vitessio/vitess/pull/10066, which is not merged yet, so there's a lot of noise in this PR. The actual diff is https://github.com/vitessio/vitess/compare/planetscale:conn-com-dump-gtid-file-pos...planetscale:vitess:endtoend-cluster-improvements?expand=1~~

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests are not required
-   [x] Documentation is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
LGTM

File: go/test/endtoend/cluster/mysqlctl_process.go [hunk 1]
```diff
@@ -215,6 +223,16 @@ func (mysqlctl *MysqlctlProcess) CleanupFiles(tabletUID int) {
 	os.RemoveAll(path.Join(os.Getenv("VTDATAROOT"), fmt.Sprintf("/vt_%010d/innodb", tabletUID)))
 }
 
+// CleanupFiles clean the mysql files to make sure we can start the same process again
```
Comment (mattlord) [inline go/test/endtoend/cluster/mysqlctl_process.go:226]: Wrong comment 🙂

**Review by mattlord — APPROVED** [review]
Just had a couple of minor comments to address whenever/however you prefer.

File: go/test/endtoend/cluster/cluster_process.go [hunk 2]
```diff
@@ -669,8 +669,8 @@ func (cluster *LocalProcessCluster) NewVtgateInstance() *VtgateProcess {
 	return vtgateProcInstance
 }
 
-// NewCluster instantiates a new cluster
-func NewCluster(cell string, hostname string) *LocalProcessCluster {
+// NewBareCluster instantiates a new cluster and does not assume existence of any of the binaries
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/test/endtoend/cluster/cluster_process.go]: Changes to `processes`

File: go/test/endtoend/cluster/mysqlctl_process.go (hunk 1 above)
Comment (shlomi-noach) [reply go/test/endtoend/cluster/mysqlctl_process.go]: fixed

**Comment by shlomi-noach** [comment]
having fixed requested issues I'm now merging.

**Pull request merged**